In [9]:
library(readxl)
library(tidyverse)
library(nnet)
library(lmtest)

In [10]:
# Read the Excel file
data <- read_excel("./data/exercice1.xlsx")
# Display the first few rows of the data
head(data)
str(data)

ID sexe Region Age etat_civil N_educatif Membre_famille Revenu_famille
1 1  0    1      2   1          3           5             0             
2 2  0    1      4   2          3           5             0             
3 3  0    1      4   2          3          11             0             
4 5  0    1      3   2          3           4             0             
5 8  0    1      1   1          3           5             1             
6 9  0    1      4   3          3           4             1             
  Revenu_Mensuelle Redoublant ⋯ Niveau_educ_pere Niveau_educ_mere Sign_ecole
1   0              1          ⋯ 2                2                1         
2   0              1          ⋯ 2                1                1         
3 100              1          ⋯ 2                1                2         
4   0              1          ⋯ 1                1                1         
5   0              1          ⋯ 3                3                3         
6 100              0          ⋯ 2                3                4         
  Endroit_freq Ph_famille Ricule_Prof Ricule_collegue Sensation_apres_ecole
1 1            1          0           1               1                    
2 2            2          1           1               1                    
3 1            3          0           0               1                    
4 1            2          1           1               3                    
5 4            4          0           0               3                    
6 1            4          1           1               1                    
  Apres_repture_ecole Violence
1 1                   1       
2 1                   1       
3 2                   0       
4 3                   1       
5 2                   1       
6 2                   0

tibble [91 × 24] (S3: tbl_df/tbl/data.frame)
 $ ID                   : num [1:91] 1 2 3 5 8 9 10 11 12 15 ...
 $ sexe                 : num [1:91] 0 0 0 0 0 0 0 0 0 0 ...
 $ Region               : num [1:91] 1 1 1 1 1 1 1 1 1 1 ...
 $ Age                  : num [1:91] 2 4 4 3 1 4 4 4 4 4 ...
 $ etat_civil           : num [1:91] 1 2 2 2 1 3 3 2 2 2 ...
 $ N_educatif           : num [1:91] 3 3 3 3 3 3 3 3 3 3 ...
 $ Membre_famille       : num [1:91] 5 5 11 4 5 4 7 8 9 10 ...
 $ Revenu_famille       : num [1:91] 0 0 0 0 1 1 0 0 0 0 ...
 $ Revenu_Mensuelle     : num [1:91] 0 0 100 0 0 100 100 0 0 0 ...
 $ Redoublant           : num [1:91] 1 1 1 1 1 0 1 1 1 1 ...
 $ Prob_Enseignat       : num [1:91] 0 1 0 1 0 0 0 1 0 0 ...
 $ Aide_ecole           : num [1:91] 1 4 4 4 1 4 5 1 5 2 ...
 $ Rq_Comprtement       : num [1:91] 1 4 1 2 1 1 1 1 1 1 ...
 $ Relat_Parent         : num [1:91] 1 1 1 1 1 1 1 1 3 1 ...
 $ Niveau_educ_pere     : num [1:91] 2 2 2 1 3 2 2 1 2 2 ...
 $ Niveau_educ_mere     : nu

## Expliquer la variable après rupture école

In [12]:
model1 <- multinom(Apres_repture_ecole ~ sexe + Age + Region + etat_civil
                   + N_educatif + Membre_famille + Revenu_famille
                   + Revenu_Mensuelle + Redoublant + Prob_Enseignat
                   + Aide_ecole + Rq_Comprtement + Relat_Parent
                   + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole
                   + Endroit_freq + Ph_famille + Ricule_Prof
                   + Ricule_collegue, data = data)
summary(model1)

# weights:  88 (63 variable)


initial  value 124.766493 
iter  10 value 78.622797
iter  20 value 63.239268
iter  30 value 55.258599
iter  40 value 49.846960
iter  50 value 47.101855
iter  60 value 43.719936
iter  70 value 43.481027
iter  80 value 43.465618
iter  90 value 43.465149
final  value 43.465144 
converged


Warning message in sqrt(diag(vc)):
"NaNs produced"


Call:
multinom(formula = Apres_repture_ecole ~ sexe + Age + Region + 
    etat_civil + N_educatif + Membre_famille + Revenu_famille + 
    Revenu_Mensuelle + Redoublant + Prob_Enseignat + Aide_ecole + 
    Rq_Comprtement + Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + 
    Sign_ecole + Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = data)

Coefficients:
  (Intercept)      sexe         Age       Region  etat_civil    N_educatif
2   -7.489772 245.87263  -0.4376716   -2.4084838    4.901879 -5.771419e-01
3  -10.469941  47.08085   0.7066992    0.6411367    5.941687  8.525498e-03
4 -611.468711  82.99164 208.3453744 -225.7999524 -128.520080 -1.605677e+02
  Membre_famille Revenu_famille Revenu_Mensuelle   Redoublant Prob_Enseignat
2      0.2134464      0.4469855       0.00356250    0.6462154      1.3745384
3      0.1245541      0.7613484       0.00124591   -0.2674273     -0.3408284
4     23.4909023    -32.9186956       0.42818081 -187.7576202    -38.2412999
  Aide

## Facteurs influençant la rupture scolaire

- **Sexe** :  
  L'impact du sexe est significatif, avec un coefficient de 245.87 pour la catégorie masculine, indiquant que ce facteur a une influence notable. Toutefois, la grande valeur du coefficient pourrait aussi signaler des problèmes de colinéarité avec d'autres variables.

- **Âge** :  
  L'âge a un effet variable, avec un coefficient de -0.44 pour chaque année supplémentaire. Cela suggère que, selon les catégories, l'augmentation de l'âge pourrait réduire légèrement la probabilité de rupture scolaire.

- **Région** :  
  La région semble avoir une influence, mais avec un coefficient négatif de -2.41. Cela pourrait refléter des disparités d'accès à l'éducation selon la zone géographique.

- **État civil et niveau éducatif des parents** :  
  L'état civil a un effet protecteur : un statut marital stable réduit la probabilité de rupture scolaire. Le niveau éducatif des parents (coefficients autour de 0.41 pour l'éducation du père et 0.99 pour la mère) montre qu’un niveau d'éducation plus élevé est lié à un moindre risque de rupture scolaire.

- **Revenu familial et mensuel** :  
  Un revenu familial plus élevé (coefficients de 0.45 et 0.0045 respectivement pour le revenu familial et mensuel) diminue la probabilité d'abandonner l'école, ce qui confirme l'influence des conditions économiques sur l'accès à l'éducation.

- **Redoublement** :  
  Le redoublement a un effet positif, avec un coefficient de 0.65, ce qui indique que les élèves ayant redoublé sont plus susceptibles de quitter l'école.

- **Probabilité d'un enseignant** :  
  La qualité de l'enseignement, représentée par la probabilité d'un enseignant, semble avoir un impact important avec un coefficient de 1.37, suggérant qu'un enseignement de qualité réduit les risques de rupture scolaire.

- **Aide à l'école** :  
  L’existence d’une aide scolaire est protectrice, avec un coefficient de 0.52, réduisant la probabilité d'abandon scolaire.

- **Comportement et relation avec les parents** :  
  Un comportement problématique (coefficient de -1.24) et une mauvaise relation avec les parents (coefficient de -0.67) augmentent significativement le risque de rupture scolaire.

- **Phénomènes familiaux** :  
  Des tensions familiales, indicatives de problèmes familiaux, augmentent aussi le risque d'abandon scolaire, avec un coefficient de 0.82.

- **Railleries des professeurs et collègues** :  
  Le harcèlement scolaire, qu'il vienne des professeurs ou des camarades, est un facteur de risque majeur pour la rupture scolaire, avec des coefficients de -1.19 et 0.82 respectivement.


## Déterminer la qualité du modèle

In [21]:
# Make predictions
predictions <- predict(model1, newdata = data)

# Confusion matrix
confusion_matrix <- table(data$Apres_repture_ecole, predictions)
print(confusion_matrix)

   predictions
     1  2  3  4
  1 29  0  0  0
  2  3 23  5  0
  3  2  5 12  1
  4  0  0  0 10


### Matrice de confusion :

- La première ligne montre que 29 prédictions ont été correctes pour la classe 1 (aucune erreur pour cette classe).
- La deuxième ligne indique que 23 prédictions étaient correctes pour la classe 2, mais 3 ont été incorrectes et ont été classées comme 1, et 5 comme 3.
- La troisième ligne montre que 12 prédictions étaient correctes pour la classe 3, mais 2 ont été incorrectes et ont été classées comme 1, 5 comme 2, et 1 comme 4.
- La quatrième ligne montre que 10 prédictions étaient correctes pour la classe 4, sans erreurs.

In [24]:
# Accuracy calculation
accuracy <- sum(diag(confusion_matrix)) / sum(confusion_matrix)
print(paste("Accuracy:", accuracy))

[1] "Accuracy: 0.822222222222222"


### Précision globale :

- *Exactitude (Accuracy) de 0.822* : Cela signifie que 82,22 % des prédictions faites par le modèle étaient correctes. C’est un bon indicateur de la performance globale du modèle.

## Expliquer la variable sensation après rupture école

In [26]:
model2 <- multinom(Sensation_apres_ecole ~ sexe + Age + Region + etat_civil
                   + N_educatif + Membre_famille + Revenu_famille
                   + Revenu_Mensuelle + Redoublant + Prob_Enseignat
                   + Aide_ecole + Rq_Comprtement + Relat_Parent
                   + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole
                   + Endroit_freq + Ph_famille + Ricule_Prof
                   + Ricule_collegue, data = data)
summary(model2)

# weights:  66 (42 variable)


initial  value 98.875106 
iter  10 value 69.921757
iter  20 value 63.545892
iter  30 value 62.811185
iter  40 value 62.710519
iter  50 value 62.693622
iter  60 value 62.691463
final  value 62.691445 
converged


Call:
multinom(formula = Sensation_apres_ecole ~ sexe + Age + Region + 
    etat_civil + N_educatif + Membre_famille + Revenu_famille + 
    Revenu_Mensuelle + Redoublant + Prob_Enseignat + Aide_ecole + 
    Rq_Comprtement + Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + 
    Sign_ecole + Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = data)

Coefficients:
  (Intercept)       sexe        Age    Region etat_civil N_educatif
2  -2.0092479 -15.288989 1.70302770  2.247275 -2.1451832 -1.1150235
3  -0.9491642   1.817137 0.05406014 -1.968661  0.4986185 -0.6205035
  Membre_famille Revenu_famille Revenu_Mensuelle Redoublant Prob_Enseignat
2    -0.06729289     -1.5735217     0.0023857860 -0.5856426     -0.1390584
3    -0.14471002     -0.3600626     0.0007416636  3.8445603      1.6831663
  Aide_ecole Rq_Comprtement Relat_Parent Niveau_educ_pere Niveau_educ_mere
2 -0.1679547     0.14149738  -0.33008611        -0.311379        0.9435157
3  0.3077777    -0.02068932   0.0

### Interprétation du modèle pour la variable "Sensation après rupture école"

- **Sexe** :
    - Impact significatif sur la sensation après rupture scolaire.
    - Coefficient négatif pour la catégorie 2 (Sensation modérée), suggérant que les hommes sont moins susceptibles de ressentir cette sensation par rapport aux femmes.

- **Âge** :
    - L'âge a un effet positif sur la catégorie 2 (Sensation modérée), mais un effet faible sur la catégorie 3 (Sensation forte).

- **Région** :
    - L'effet est négatif pour la catégorie 3 (Sensation forte), indiquant que certaines régions sont associées à une moindre probabilité de ressentir cette sensation.

- **État civil** :
    - Un statut marital stable a un effet protecteur contre la sensation modérée ou forte après la rupture scolaire.

- **Niveau éducatif des parents** :
    - Un niveau éducatif plus élevé de la mère augmente la probabilité de la catégorie 3 (Sensation forte).

- **Revenu familial et mensuel** :
    - Un revenu plus élevé réduit la probabilité de ressentir une sensation modérée ou forte après la rupture scolaire.

- **Redoublement** :
    - Le redoublement augmente la probabilité de ressentir une sensation forte (catégorie 3).

- **Probabilité d'un enseignant** :
    - La qualité de l'enseignement a un effet protecteur contre la sensation modérée.

- **Aide à l'école** :
    - L'aide scolaire réduit la probabilité de sensation modérée et forte après la rupture scolaire.

- **Comportement et relation avec les parents** :
    - Un bon comportement et une relation positive avec les parents diminuent la probabilité de ressentir une sensation forte.

- **Phénomènes familiaux** :
    - Des tensions familiales augmentent la probabilité de la sensation modérée.

- **Railleries des professeurs et collègues** :
    - Le harcèlement scolaire a un effet majeur, augmentant la probabilité de ressentir une sensation forte.

## Déterminer la qualité du modèle

In [27]:
# Make predictions
predictions2 <- predict(model2, newdata = data)

# Confusion matrix
confusion_matrix2 <- table(data$Sensation_apres_ecole, predictions2)
print(confusion_matrix2)

   predictions2
     1  2  3
  1 21  4  7
  2  3 23  6
  3  3  6 17


### Matrice de confusion :

- La première ligne montre que 21 prédictions ont été correctes pour la classe 1, avec 4 erreurs classées comme 2 et 7 comme 3.
- La deuxième ligne indique que 23 prédictions étaient correctes pour la classe 2, mais 3 ont été incorrectes et ont été classées comme 1, et 6 comme 3.
- La troisième ligne montre que 17 prédictions étaient correctes pour la classe 3, mais 3 ont été incorrectes et ont été classées comme 1, et 6 comme 2.


In [28]:
# Accuracy calculation
accuracy2 <- sum(diag(confusion_matrix2)) / sum(confusion_matrix2)
print(paste("Accuracy:", accuracy2))

[1] "Accuracy: 0.677777777777778"


### Précision globale :

- *Exactitude (Accuracy) de 0.677* : Cela signifie que le modèle a correctement classé environ 68 % des observations. Il y a encore des erreurs de classification, surtout entre les classes 1 et 3, ainsi qu'entre les classes 2 et 3. Cependant, avec une précision de 67,78 %, le modèle peut être considéré comme acceptable.

### Expliquer la variable violence

In [29]:
model3 <- multinom(Violence  ~ sexe + Age + Region + etat_civil
                   + N_educatif + Membre_famille + Revenu_famille
                   + Revenu_Mensuelle + Redoublant + Prob_Enseignat
                   + Aide_ecole + Rq_Comprtement + Relat_Parent
                   + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole
                   + Endroit_freq + Ph_famille + Ricule_Prof
                   + Ricule_collegue, data = data)
summary(model3)

# weights:  22 (21 variable)
initial  value 62.383246 
iter  10 value 36.964839
iter  20 value 34.414180
final  value 34.343438 
converged


Call:
multinom(formula = Violence ~ sexe + Age + Region + etat_civil + 
    N_educatif + Membre_famille + Revenu_famille + Revenu_Mensuelle + 
    Redoublant + Prob_Enseignat + Aide_ecole + Rq_Comprtement + 
    Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole + 
    Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = data)

Coefficients:
                       Values   Std. Err.
(Intercept)       6.068413312 3.713088928
sexe              1.524809931 2.123160842
Age              -1.055798304 0.533031905
Region            1.002753591 1.025679325
etat_civil        0.023586691 0.854480858
N_educatif       -0.673341017 0.624908213
Membre_famille   -0.061502670 0.193547317
Revenu_famille    1.282931321 0.974624342
Revenu_Mensuelle -0.004960213 0.002147526
Redoublant       -1.211504998 0.971903269
Prob_Enseignat   -2.011345947 0.982459715
Aide_ecole       -0.196828062 0.270754003
Rq_Comprtement    0.509204931 0.424193473
Relat_Parent      0.418167382 0.67160

### Interprétation des variables :

- **Intercept** : 6.068 (indique une valeur de base élevée pour la violence en l'absence des autres variables).
- **sexe** : 1.525 (le sexe a un impact positif sur la violence, mais l'écart-type élevé indique une incertitude).
- **Age** : -1.056 (plus l'âge augmente, plus la violence diminue).
- **Region** : 1.003 (la région a un effet positif sur la violence, mais l'incertitude est importante).
- **etat_civil** : 0.024 (l'état civil a un effet négligeable sur la violence).
- **N_educatif** : -0.673 (un niveau éducatif plus élevé réduit la violence, mais avec une incertitude relativement élevée).
- **Membre_famille** : -0.062 (l'effet de la présence d'un membre de famille est faible).
- **Revenu_famille** : 1.283 (des revenus familiaux plus élevés augmentent la violence).
- **Revenu_Mensuelle** : -0.005 (l'impact est faible, avec une faible variation du revenu mensuel).
- **Redoublant** : -1.212 (les redoublants ont un score de violence plus faible).
- **Prob_Enseignat** : -2.011 (une probabilité élevée d'un enseignant réduit la violence).
- **Aide_ecole** : -0.197 (l'impact de l'aide scolaire est faible).
- **Rq_Comprtement** : 0.509 (un comportement problématique augmente la violence).
- **Relat_Parent** : 0.418 (une relation moins bonne avec les parents augmente la violence).
- **Niveau_educ_pere** : -0.504 (le niveau éducatif du père a un léger effet négatif sur la violence).
- **Niveau_educ_mere** : -0.204 (le niveau éducatif de la mère a un léger effet négatif sur la violence).
- **Sign_ecole** : -0.094 (l'impact de la signature de l'école est faible).
- **Endroit_freq** : 0.392 (l'endroit fréquenté augmente légèrement la violence).
- **Ph_famille** : -1.159 (des phénomènes familiaux réduisent la violence).
- **Ricule_Prof** : 2.830 (les railleries des professeurs augmentent fortement la violence).
- **Ricule_collegue** : 0.984 (les railleries des camarades augmentent également la violence).
